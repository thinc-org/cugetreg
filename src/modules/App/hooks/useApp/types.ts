import { Color } from '@material-ui/core/Alert'

export interface UseAppProps {
  open: boolean
  message: string
  action?: string
  emitMessage: (messageText: string, type: Color, actiontText?: string) => void
  handleClose: (_: unknown, reason: string) => void
  messageType: Color
  actionText?: string
}
