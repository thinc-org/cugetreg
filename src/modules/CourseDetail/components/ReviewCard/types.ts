import { Review } from '@/common/types/reviews'

export interface ReviewCardProps extends Review {}
