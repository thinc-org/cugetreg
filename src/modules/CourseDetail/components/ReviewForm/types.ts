export interface ReviewFormProps {
  courseNo: string
}
