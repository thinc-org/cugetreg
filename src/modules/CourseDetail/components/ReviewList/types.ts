import { Course } from '@thinc-org/chula-courses'

export interface ReviewListProps {
  course: Course
}
