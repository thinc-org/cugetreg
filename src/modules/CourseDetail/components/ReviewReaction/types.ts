export interface ReviewReactionProps {
  type: ReviewReactionType
  defaultPressed: boolean
  reactionCount: number
}

export enum ReviewReactionType {
  Like = 'like',
  Dislike = 'dislike',
}
