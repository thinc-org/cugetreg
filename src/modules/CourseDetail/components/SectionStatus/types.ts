import { TypographyProps } from '@material-ui/core'
import { Capacity } from '@thinc-org/chula-courses'

export type AvailableStatus = 'avialable' | 'full' | 'closed'
export interface SectionStatusProps extends TypographyProps {
  capacity: Capacity
  closed: boolean
}
