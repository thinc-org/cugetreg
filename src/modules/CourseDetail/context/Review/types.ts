import { TNode } from '@udecode/plate-core'

import { Review, ReviewInteractionType } from '@/common/types/reviews'

export type ReviewState = Pick<Review, 'rating' | 'academicYear' | 'semester'> & {
  content: TNode[]
}

export interface ReviewContextValues {
  reviews: Review[]
  myPendingReviews: Review[]
  setInteraction: (reviewId: string, interaction: ReviewInteractionType) => void
  reportReview: (reviewId: string) => void
  deleteMyReview: (reviewId: string) => void
  editMyReview: (reviewId: string) => void
  cancelEditReview: () => void
  submitReview: () => void
  submitEditedReview: (reviewId: string) => void
  editingReviewId?: string
}

export interface ReviewProviderProps {
  courseNo: string
  initialReviews: Review[]
}
