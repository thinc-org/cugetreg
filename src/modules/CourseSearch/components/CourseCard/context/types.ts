import { Course, DayOfWeek, Capacity, Section } from '@thinc-org/chula-courses'

export interface CourseCardContextValue {
  course: Course
  isGenEd: boolean
  classDays: DayOfWeek[]
  courseCapacity: Capacity
  sectionNumbers: string[]
  selectedSectionNumber: string
  setSectionNumber: (sectionNo: string) => void
  selectedSection: Section
  teachers: string[]
  isExpanded: boolean
  onToggle: () => void
}
