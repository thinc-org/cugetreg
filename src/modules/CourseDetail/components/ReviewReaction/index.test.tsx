import { shallow } from 'enzyme'
import { MdThumbUpOffAlt, MdThumbUp, MdThumbDownOffAlt, MdThumbDown } from 'react-icons/md'

import { ReviewReactionProps, ReviewReactionType } from './types'

describe('ReviewReaction', () => {
  const mockProps: ReviewReactionProps = {
    type: ReviewReactionType.Like,
    defaultPressed: false,
    reactionCount: 42,
  }

  it.each`
    type                          | defaultPressed | expectedIcon
    ${ReviewReactionType.Like}    | ${false}       | ${MdThumbUpOffAlt}
    ${ReviewReactionType.Dislike} | ${false}       | ${MdThumbDownOffAlt}
    ${ReviewReactionType.Like}    | ${true}        | ${MdThumbUp}
    ${ReviewReactionType.Dislike} | ${true}        | ${MdThumbDown}
  `(
    'should render ReviewReaction type $type like correctly when defaultPressed=$defaultPressed ',
    async ({ type, expectedIcon, defaultPressed }) => {
      const { ReviewReaction } = await import('.')
      const wrapper = shallow(<ReviewReaction {...mockProps} type={type} defaultPressed={defaultPressed} />)
      expect(wrapper.find(expectedIcon)).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    }
  )

  it.each`
    type                          | expectedUnpressedIcon | expectedPressedIcon
    ${ReviewReactionType.Like}    | ${MdThumbUpOffAlt}    | ${MdThumbUp}
    ${ReviewReactionType.Dislike} | ${MdThumbDownOffAlt}  | ${MdThumbDown}
  `(
    'should change icon when the component was clicked, type=$type',
    async ({ type, expectedUnpressedIcon, expectedPressedIcon }) => {
      const { ReviewReaction } = await import('.')
      const wrapper = shallow(<ReviewReaction {...mockProps} type={type} />)
      expect(wrapper.find(expectedUnpressedIcon)).toBeTruthy()
      wrapper.simulate('click')
      expect(wrapper.find(expectedPressedIcon)).toBeTruthy()
      wrapper.simulate('click')
      expect(wrapper.find(expectedUnpressedIcon)).toBeTruthy()
    }
  )
})
