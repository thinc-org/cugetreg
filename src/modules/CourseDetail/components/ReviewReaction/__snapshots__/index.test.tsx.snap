// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewReaction should render ReviewReaction type dislike like correctly when defaultPressed=false  1`] = `
<ForwardRef(Stack)
  alignItems="center"
  direction="row"
  onClick={[Function]}
  spacing={1}
>
  <SelectedIcon />
  <ForwardRef(Typography)
    color="primaryRange.50"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Stack)>
`;

exports[`ReviewReaction should render ReviewReaction type dislike like correctly when defaultPressed=true  1`] = `
<ForwardRef(Stack)
  alignItems="center"
  direction="row"
  onClick={[Function]}
  spacing={1}
>
  <SelectedIcon />
  <ForwardRef(Typography)
    color="primaryRange.50"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Stack)>
`;

exports[`ReviewReaction should render ReviewReaction type like like correctly when defaultPressed=false  1`] = `
<ForwardRef(Stack)
  alignItems="center"
  direction="row"
  onClick={[Function]}
  spacing={1}
>
  <SelectedIcon />
  <ForwardRef(Typography)
    color="primaryRange.50"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Stack)>
`;

exports[`ReviewReaction should render ReviewReaction type like like correctly when defaultPressed=true  1`] = `
<ForwardRef(Stack)
  alignItems="center"
  direction="row"
  onClick={[Function]}
  spacing={1}
>
  <SelectedIcon />
  <ForwardRef(Typography)
    color="primaryRange.50"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Stack)>
`;
